### PR TITLE
Edit Contentlet: Allow User Write Code

### LINKED ISSUE #26045: Edit Contentlet: Allow User Write Code
### Parent Issue

#25445 

### Task

As a dotCMS user, I desire the capability to write code directly in the binary field.

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

dotCMS users should have the ability to write code directly instead of being required to upload a file.

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

- **Fima Design**: https://www.figma.com/file/YMTdSO5gKf18rc1pCCv1jN/Edit-Contentlet?type=design&node-id=479%3A17595&mode=design&t=cOHbia6V2I3zTbcp-1

- **Prototype**: https://www.figma.com/proto/YMTdSO5gKf18rc1pCCv1jN/Edit-Contentlet?node-id=365-14264&scaling=scale-down&page-id=365%3A14229&starting-point-node-id=365%3A14264&mode=design&t=dBd4KhDyJmk0nuPC-1

Binary Field Write Code

### Needs
- Required file name field
- Configure the **syntax highlighting** of the Monaco editor based on the extension (file name) entered by the user.

### Assumptions
- When a user "saves" the code, we need to write the code out to a temp file using the tempAPI. Then, when we finally save the content, we reference the temp file for the binary field, which will get picked up.

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by rjvelazco**
### <samp>🤖 Generated by Copilot at 80968f2</samp>

This pull request adds a new component `DotBinaryFieldEditorComponent` that allows users to edit binary files with the Monaco code editor in the binary field type. It also modifies the `DotBinaryFieldComponent` and its related files to use the new component and improve the user experience and functionality.

## Related Issue
Fixes #26045

# Explanation of Changes

### <samp>🤖 Generated by Copilot at 80968f2</samp>

*  Import and register MonacoEditorModule for the app ([link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-07240e1adf9e007eb5e1201d172f486673b52f09fef6bd4383375b44b7ab3769R1-R2), [link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-07240e1adf9e007eb5e1201d172f486673b52f09fef6bd4383375b44b7ab3769L25-R27))
* Add dot-binary-field-editor component to render the editor and the form in the binary field component ([link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-fe43ae6cefd137f07a1883a5f279da1a1775ea58561fcc1f770a3a617d65d40eL92-R93), [link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-740e087d4ab3dc50c530488107c47d9fd0da08096e92f5459e56698387a23554R32), [link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-740e087d4ab3dc50c530488107c47d9fd0da08096e92f5459e56698387a23554R66), [link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-a5971decf72277b0923f8fc6c54ef58c3d2cb6148015a72e98624f2f9b79c728R1-R44), [link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-2b9cc72ba966b6a66f6bf0bd2a5f75ff9261650e11c6ac9a28b55d5b415cec54R1-R55), [link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-b4a4eb20abef254ef786f5dc2709450ead6ddb107d707a32503288ae4a9f774aR1-R151))
* Prevent dialog from closing on escape key in the binary field component ([link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-fe43ae6cefd137f07a1883a5f279da1a1775ea58561fcc1f770a3a617d65d40eR82))
* Simulate successful upload in the binary field component stories ([link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-a939da2218f476e5f8f219886765fa9fc0ceeaa94fe280e8a75a04d9c3886768L53-R64))
* Add output event emitter for the temp file in the dot-binary-field-editor component ([link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-b4a4eb20abef254ef786f5dc2709450ead6ddb107d707a32503288ae4a9f774aR1-R151))
* Use view child reference for the editor component in the dot-binary-field-editor component ([link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-b4a4eb20abef254ef786f5dc2709450ead6ddb107d707a32503288ae4a9f774aR1-R151))
* Create form group for the form controls in the dot-binary-field-editor component ([link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-a5971decf72277b0923f8fc6c54ef58c3d2cb6148015a72e98624f2f9b79c728R1-R44), [link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-b4a4eb20abef254ef786f5dc2709450ead6ddb107d707a32503288ae4a9f774aR1-R151))
* Configure editor options and mime type in the dot-binary-field-editor component ([link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-b4a4eb20abef254ef786f5dc2709450ead6ddb107d707a32503288ae4a9f774aR1-R151))
* Implement OnInit and AfterViewInit hooks in the dot-binary-field-editor component ([link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-b4a4eb20abef254ef786f5dc2709450ead6ddb107d707a32503288ae4a9f774aR1-R151))
* Define methods for setting editor language, uploading file, updating editor options, and enabling and disabling editor in the dot-binary-field-editor component ([link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-b4a4eb20abef254ef786f5dc2709450ead6ddb107d707a32503288ae4a9f774aR1-R151))
* Add unit test for the dot-binary-field-editor component ([link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-df86fd17a4a03cd49192a68c074f2bf7ce8054b0efda77b49af7b4c5a124de43R1-R22))
* Add mock message for the save button label in the dot-binary-field-editor component ([link](https://github.com/dotCMS/core/pull/26426/files?diff=unified&w=0#diff-782111a03efe958f664584c0f4b51b86faada086ea84867dc6b367f1265f8b44L22-R23))

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

**Review by zJaaal — APPROVED** [review]

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.html [hunk 1]
```diff
@@ -79,19 +79,26 @@
...
+        [styleClass]="isEditorMode(vm.mode) ? 'screen-cover' : ''"
         (onHide)="afterDialogClose()">
-        <div [ngSwitch]="vm.mode">
-            <div *ngSwitchCase="BINARY_FIELD_MODE.URL" data-testId="url">
+        <ng-container [ngSwitch]="vm.mode">
+            <ng-container *ngSwitchCase="BINARY_FIELD_MODE.URL" data-testId="url">
                 <dot-dot-binary-field-url-mode
                     [accept]="accept"
                     [maxFileSize]="maxFileSize"
                     (tempFileUploaded)="setTempFile($event)"
```
Comment (fmontes) [inline core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.html:90]:
```suggestion
        <ng-container [ngSwitch]="vm.mode">
                <dot-dot-binary-field-url-mode *ngSwitchCase="BINARY_FIELD_MODE.URL" data-testId="url"
                    [accept]="accept"
                    [maxFileSize]="maxFileSize"
                    (tempFileUploaded)="setTempFile($event)"
```

**Review by fmontes — APPROVED** [review]

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.stories.ts [hunk 2]
```diff
@@ -78,7 +75,7 @@ export default {
         })
     ],
     args: {
-        accept: ['image/*'],
+        accept: ['image/*', '.ts'],
```
Comment (jdotcms) [inline core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.stories.ts:78]: text plain? js, xml?

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.html (hunk 1 above)
Comment (rjvelazco) [reply core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.html]: Done!

**Comment by fmontes** [comment]
Sonar Qube is broken.

**Pull request merged**